crypto/evp/evp_rand.c

```diff
@@ -333,12 +333,6 @@ EVP_RAND_CTX *EVP_RAND_CTX_new(EVP_RAND *rand, EVP_RAND_CTX *parent)
         return NULL;
     }
     if (parent != NULL) {
-        if (!EVP_RAND_enable_locking(parent)) {
```

Comment:
Removing this block seems wrong.  The parent needs to support locking for a child to safely use it.

Do you remember the rationale behind removing this?

Reply:
Yes. Because setting locking at this point is too late. The parent needs to be configured for locking *before* the children are created. Otherwise, if there are multiple children in different threads, the children race trying to create the parent lock.

Reply:
For our own DRBGs, we instead move the lock creation to where the parent is created (see `RAND_get0_primary`).

Reply:
Thanks for the explanation.  I'll raise an issue as a reminder to update the documentation for `EVP_RAND_CTX_new` to note that the parent should have locking enabled before creating the child.

Reply:
It's possible that there is a similar race in `ossl_drbg_enable_locking()`:

https://github.com/openssl/openssl/blob/3d63348a871d2319f7ff3512f97fd660fa7fadea/providers/implementations/rands/drbg.c#L727-L744

The solution could be the same.